code/game/objects/items/weapons/stunbaton.dm

```diff
@@ -132,22 +134,35 @@
 	cell = null
 	turned_on = FALSE
 	update_icon(UPDATE_ICON_STATE)
+	STOP_PROCESSING(SSobj, src)
 
 /obj/item/melee/baton/attack_self(mob/user)
 	if(cell?.charge >= hitcost)
 		turned_on = !turned_on
 		to_chat(user, "<span class='notice'>[src] is now [turned_on ? "on" : "off"].</span>")
 		playsound(src, "sparks", 75, TRUE, -1)
+		if(turned_on)
+			START_PROCESSING(SSobj, src)
+		else
+			STOP_PROCESSING(SSobj, src)
 	else
 		if(isrobot(loc))
 			to_chat(user, "<span class='warning'>You do not have enough reserve power to charge [src]!</span>")
 		else if(!cell)
 			to_chat(user, "<span class='warning'>[src] does not have a power source!</span>")
 		else
 			to_chat(user, "<span class='warning'>[src] is out of charge.</span>")
+		STOP_PROCESSING(SSobj, src)
 	update_icon()
 	add_fingerprint(user)
 
+/obj/item/melee/baton/process()
+	cell.use(100)
+	if(!cell.charge)
+		turned_on = FALSE
+		playsound(src, "sparks", 75, TRUE, -1)
+		update_icon()
+		STOP_PROCESSING(SSobj, src)
```

Comment:
Basically having these 4 lines copy pasted over the file hurts my eyes. Can you make a `turn_on` and `turn_off` proc?

Reply:
i could try this but also not too confident in coding abilities

anyways it'll be at least week until i can, going to be busy for a bit